python_modules/libraries/dagster-aws/dagster_aws/ecs/executor.py

```diff
@@ -0,0 +1,358 @@
+
+        run_task_kwargs["tags"] = [
+            *run_task_kwargs.get("tags", []),
+            *[
+                {"key": k, "value": v}
+                for k, v in step_handler_context.dagster_run.dagster_execution_info.items()
+            ],
+            {"key": "dagster/step-key", "value": step_key},
+        ]
+
+        if run.remote_job_origin:
+            run_task_kwargs["tags"] = [
+                *run_task_kwargs["tags"],
+                {
+                    "key": "dagster/code-location",
+                    "value": run.remote_job_origin.repository_origin.code_location_origin.location_name,
+                },
+            ]
+
+        task_overrides = self._get_task_overrides(step_tags) or {}
+
+        task_overrides["containerOverrides"] = task_overrides.get("containerOverrides", [])
+
+        # container name has to match since we are assuming we are using the same task
+        executor_container_name = run_launcher.get_container_name(container_context)
+        executor_env_vars = [
+            {"name": env["name"], "value": env["value"]}
+            for env in step_handler_context.execute_step_args.get_command_env()
+        ]
+
+        # inject Executor command and env vars into the container overrides
+        # if they are defined
+        # otherwise create a new container overrides for the executor container
+        for container_overrides in task_overrides["containerOverrides"]:
+            # try to update existing container overrides for the executor container
+            if container_overrides["name"] == executor_container_name:
+                if "command" in container_overrides and container_overrides["command"] != args:
+                    raise DagsterInvariantViolationError(
+                        f"The 'command' field for {executor_container_name} container is not allowed in the 'containerOverrides' field of the task overrides."
+                    )
+
+                # update environment variables & command
+                container_overrides["command"] = args
+                container_overrides["environment"] = (
+                    container_overrides.get("environment", []) + executor_env_vars
+                )
+                break
+        # if no existing container overrides for the executor container, add new container overrides
+        else:
+            task_overrides["containerOverrides"].append(
+                {
+                    "name": executor_container_name,
+                    "command": args,
+                    "environment": executor_env_vars,
+                }
+            )
+
+        if isinstance(run_launcher, EcsRunLauncher):
+            # get ECSRunLauncher run_task kwargs and override them
+            image = run_launcher.get_image_for_run(run)
+            run_task_kwargs = deep_merge_dicts(
+                run_launcher.get_run_task_kwargs(run, image, container_context), run_task_kwargs
```

Comment:
> I'm not sure off the top of my head if setting this as a tag in the launchpad would have any effect - I suspect it would not? i think it would still get overridden. This would work for checking the tag after the run was launched though

It should work. Both run launcher and executor code will (now) pick up the user-provided image. 
That's how it works with the K8s run launcher / executor.

Reply:
I only found one `add_run_tags` usage in the run launcher code and it does not set the `dagster/image` tag.
Looks like this is safe to do. 
Additionally, we should set the `dagster/image` tag when the image comes from the code location config (`DAGSTER_CURRENT_IMAGE` or cli argument)

Reply:
we decided to remove this logic